docs/architecture/adr-010-unified-cli-arguments-hermes.md

```diff
@@ -0,0 +1,185 @@
+# ADR 010: Unified approach for CLI arguments for Hermes v1.0.0
+
+## Changelog
+* 15.06.2022: Proposed.
+
+## Context
+
+In this ADR we provide recommendations and intuitions to using flags for all the arguments of the Hermes commands.
+
+The problem we are trying to solve is a unified approach to CLI arguments for Hermes v1.0.0.
+
+## Decision
+
+To avoid confusion, all the parameters should take long flags. The following general scenarios should be applied:
+
+* Only long flags are used in order to avoid having nonintuitive flags or conflicting flags.
+* Any parameter ending with `_id` should have the `_id` removed from the flag to shorten it. For example the flag for `chain_id` should only be `chain`.
+* Flags which can be shorten and still be meaningful should be shorten. This is done for `connection`, `channel` and `sequence`, which become respectively `conn`, `chan` and `seq`.
+* For the channel and connection creation CLIs, the objects at the two ends are suffixed by `-a` and `-b`.
+* All the specifiers for `chain`, `connection`, `channel` and `port` flags are suffixed. These specifiers are `host`, `reference`, `a`, `b` and `counterparty`.
+
+The following commands are implemented, with the binary name `hermes` omitted:
+
+### Hermes global flags
+
+* `hermes --config <CONFIG> <COMMAND>`
+
+* `hermes --json <COMMAND>`
+
+### Commands for clients
+
+* `create client --chain-host <CHAIN_HOST_ID> --chain-reference <CHAIN_REFERENCE_ID>`
+    * Optional: `[--clock-drift <CLOCK_DRIFT>] [--trust-threshold <TRUST_THRESHOLD>] [--trusting-period <TRUSTING_PERDIOD>]`
+
+* `update client --chain-host <CHAIN_HOST_ID> --client-host <CLIENT_HOST_ID>`
+    * Optional: `[--height <REFERENCE_HEIGHT>] [--trusted-height <REFERENCE_TRUSTED_HEIGHT>]`
+
+* `upgrade client --chain-host <CHAIN_HOST_ID> --client-host <CLIENT_HOST_ID>`
```

Comment:
Teeny tiny: How about just `--client` instead of `--client-host`? We're already specifying `--chain-host`, it's pretty clear to me that the client we're specifying lives on that chain. I also think of "host" and "reference" to apply to chains only, not clients. 

Same comment applies to `update client`.

@ancazamfir @ljoss17 WDYT?

Reply:
I agree, I'll update the `update client` and `upgrade client`.